manual/source/topic/arena.rst

```diff
@@ -1040,3 +1051,79 @@ Arena introspection and debugging
         :c:func:`mps_addr_pool`, and to find out which :term:`object
         format` describes the object at the address, use
         :c:func:`mps_addr_fmt`.
+
+
+.. index::
+   single: arena extension callbacks; introduction
+   single: extension callbacks; introduction
+   single: arena contraction callbacks; introduction
+   single: contraction callbacks; introduction
+
+.. _topic-arena-extension:
+
+Arena extension callbacks
+-------------------------
+
+There are situations in which the :term:`client program` needs to be
+informed about the chunks of address space that an :term:`arena` is
+managing. To support this, the MPS allows the client program to
+specify two callback functions when creating a :term:`virtual memory
+arena`: one function is called when the arena is *extended* (that is,
+when it acquires a new chunk of address space from the operating
+system), and the other when the arena is *contracted* (that is, when
+it returns a chunk of address space to the operating system).
```

Comment:
Nm. Are the contraction calls 1:1 with the extension calls?  Do we guarantee that?  I think we probably should.  rule.generic.complete.

Reply:
Ignore: Already fixed in 74d77c80f639f384f799506fc19c2dc15f09b94b